// react-native-slider-intro/lib/commonjs/components/DotContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const DotContainer = ({
  navContainerMaxSize,
  arrayOfSlideIndex,
  fixDotBackgroundColor,
  fixDotOpacity,
  dotWidth,
  _moveSlideDotX,
  _moveSlideDotMarginX,
  animatedDotBackgroundColor
}) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.dotMainContainer, {
      maxWidth: navContainerMaxSize
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.mainDotContainer, {
      maxWidth: navContainerMaxSize
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.mainDotInnerContainer
  }, arrayOfSlideIndex.map(item => {
    return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
      key: item,
      style: [styles.fixDot, {
        backgroundColor: fixDotBackgroundColor,
        opacity: fixDotOpacity,
        width: dotWidth,
        maxWidth: dotWidth,
        height: dotWidth,
        maxHeight: dotWidth
      }]
    });
  }))), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.animatedDotContainer]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.animatedDotInnerContainer
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Animated.View, {
    style: [styles.animatedDot, {
      width: _moveSlideDotX,
      maxWidth: _moveSlideDotX,
      height: dotWidth,
      maxHeight: dotWidth,
      marginLeft: _moveSlideDotMarginX,
      backgroundColor: animatedDotBackgroundColor
    }]
  }))));
};

const styles = _reactNative.StyleSheet.create({
  dotMainContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  mainDotContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  animatedDotContainer: {
    position: 'absolute',
    flex: 1,
    flexDirection: 'row'
  },
  mainDotInnerContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  fixDot: {
    flex: 1,
    borderRadius: 100
  },
  animatedDotInnerContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  animatedDot: {
    flex: 1,
    borderRadius: 100
  }
});

var _default = DotContainer;
exports.default = _default;
