// react-native-slider-intro/lib/commonjs/components/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Item = ({
  index,
  backgroundColor,
  title,
  text,
  link,
  activeLanguage,
  image,
  slideMaxHeightPercent
}) => {
  const deviceMaxHeight = _reactNative.Dimensions.get('screen').height;

  const language = activeLanguage || 'en';
  const slideHeight = deviceMaxHeight * (slideMaxHeightPercent || 0.78);
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    key: index,
    style: [styles.slide, {
      backgroundColor
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.container, {
      height: slideHeight,
      maxHeight: slideHeight
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: styles.title
  }, Array.isArray(title) ? title[language] : title), image && /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    style: styles.image,
    source: image
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View, null, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: styles.text
  }, Array.isArray(text) ? text[language] : text), link && /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    onPress: () => _reactNative.Linking.openURL(link)
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: styles.link
  }, link)))));
};

var _default = Item;
exports.default = _default;

const styles = _reactNative.StyleSheet.create({
  container: {
    marginTop: 10,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    maxHeight: _reactNative.Dimensions.get('window').height * 0.85
  },
  slide: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  title: {
    fontSize: 24,
    color: 'white',
    textAlign: 'center'
  },
  text: {
    marginLeft: 20,
    marginRight: 20,
    lineHeight: 24,
    fontSize: 15,
    color: 'white',
    textAlign: 'center'
  },
  image: {
    maxWidth: _reactNative.Dimensions.get('window').width,
    maxHeight: _reactNative.Dimensions.get('window').width
  },
  link: {
    marginLeft: 20,
    marginRight: 20,
    fontSize: 15,
    color: '#2f39ff',
    textDecorationLine: 'underline',
    textAlign: 'center'
  }
});
