// react-native-slider-intro/lib/commonjs/components/NextContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const NextContainer = ({
  goToNewSlide,
  slide,
  setSlide,
  numberOfSlide,
  onDone,
  navContainerMaxSize,
  dotWidth,
  deviceMaxWidth,
  renderNextButton,
  renderDoneButton,
  isLastSlide,
  buttonsMaxSize
}) => {
  const {
    active,
    animations
  } = slide;
  const {
    _opacityOfNextButton,
    _opacityOfDoneButton
  } = animations;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.buttonContainer, {
      maxWidth: buttonsMaxSize
    }]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    onPress: () => goToNewSlide(active + 1, slide, setSlide, numberOfSlide, onDone, navContainerMaxSize, dotWidth, deviceMaxWidth)
  }, !isLastSlide ? /*#__PURE__*/_react.default.createElement(_reactNative.Animated.View, {
    style: {
      opacity: _opacityOfNextButton
    }
  }, renderNextButton()) : /*#__PURE__*/_react.default.createElement(_reactNative.Animated.View, {
    style: {
      opacity: _opacityOfDoneButton
    }
  }, renderDoneButton())));
};

const styles = _reactNative.StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

var _default = NextContainer;
exports.default = _default;
