// react-native-slider-intro/lib/module/components/DotContainer.js
import React from 'react';
import { View, Animated, StyleSheet } from 'react-native';

const DotContainer = ({
  navContainerMaxSize,
  arrayOfSlideIndex,
  fixDotBackgroundColor,
  fixDotOpacity,
  dotWidth,
  _moveSlideDotX,
  _moveSlideDotMarginX,
  animatedDotBackgroundColor
}) => {
  return /*#__PURE__*/React.createElement(View, {
    style: [styles.dotMainContainer, {
      maxWidth: navContainerMaxSize
    }]
  }, /*#__PURE__*/React.createElement(View, {
    style: [styles.mainDotContainer, {
      maxWidth: navContainerMaxSize
    }]
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.mainDotInnerContainer
  }, arrayOfSlideIndex.map(item => {
    return /*#__PURE__*/React.createElement(View, {
      key: item,
      style: [styles.fixDot, {
        backgroundColor: fixDotBackgroundColor,
        opacity: fixDotOpacity,
        width: dotWidth,
        maxWidth: dotWidth,
        height: dotWidth,
        maxHeight: dotWidth
      }]
    });
  }))), /*#__PURE__*/React.createElement(View, {
    style: [styles.animatedDotContainer]
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.animatedDotInnerContainer
  }, /*#__PURE__*/React.createElement(Animated.View, {
    style: [styles.animatedDot, {
      width: _moveSlideDotX,
      maxWidth: _moveSlideDotX,
      height: dotWidth,
      maxHeight: dotWidth,
      marginLeft: _moveSlideDotMarginX,
      backgroundColor: animatedDotBackgroundColor
    }]
  }))));
};

const styles = StyleSheet.create({
  dotMainContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  mainDotContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  animatedDotContainer: {
    position: 'absolute',
    flex: 1,
    flexDirection: 'row'
  },
  mainDotInnerContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  fixDot: {
    flex: 1,
    borderRadius: 100
  },
  animatedDotInnerContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  animatedDot: {
    flex: 1,
    borderRadius: 100
  }
});
export default DotContainer;
