// react-native-slider-intro/lib/module/components/Item.js
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Image, Linking, Dimensions } from 'react-native';

const Item = ({
  index,
  backgroundColor,
  title,
  text,
  link,
  activeLanguage,
  image,
  slideMaxHeightPercent
}) => {
  const deviceMaxHeight = Dimensions.get('screen').height;
  const language = activeLanguage || 'en';
  const slideHeight = deviceMaxHeight * (slideMaxHeightPercent || 0.78);
  return /*#__PURE__*/React.createElement(View, {
    key: index,
    style: [styles.slide, {
      backgroundColor
    }]
  }, /*#__PURE__*/React.createElement(View, {
    style: [styles.container, {
      height: slideHeight,
      maxHeight: slideHeight
    }]
  }, /*#__PURE__*/React.createElement(Text, {
    style: styles.title
  }, Array.isArray(title) ? title[language] : title), image && /*#__PURE__*/React.createElement(Image, {
    style: styles.image,
    source: image
  }), /*#__PURE__*/React.createElement(View, null, /*#__PURE__*/React.createElement(Text, {
    style: styles.text
  }, Array.isArray(text) ? text[language] : text), link && /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => Linking.openURL(link)
  }, /*#__PURE__*/React.createElement(Text, {
    style: styles.link
  }, link)))));
};

export default Item;
const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    maxHeight: Dimensions.get('window').height * 0.85
  },
  slide: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  title: {
    fontSize: 24,
    color: 'white',
    textAlign: 'center'
  },
  text: {
    marginLeft: 20,
    marginRight: 20,
    lineHeight: 24,
    fontSize: 15,
    color: 'white',
    textAlign: 'center'
  },
  image: {
    maxWidth: Dimensions.get('window').width,
    maxHeight: Dimensions.get('window').width
  },
  link: {
    marginLeft: 20,
    marginRight: 20,
    fontSize: 15,
    color: '#2f39ff',
    textDecorationLine: 'underline',
    textAlign: 'center'
  }
});
