// react-native-slider-intro/lib/module/components/NextContainer.js
import React from 'react';
import { View, TouchableOpacity, Animated, StyleSheet } from 'react-native';

const NextContainer = ({
  goToNewSlide,
  slide,
  setSlide,
  numberOfSlide,
  onDone,
  navContainerMaxSize,
  dotWidth,
  deviceMaxWidth,
  renderNextButton,
  renderDoneButton,
  isLastSlide,
  buttonsMaxSize
}) => {
  const {
    active,
    animations
  } = slide;
  const {
    _opacityOfNextButton,
    _opacityOfDoneButton
  } = animations;
  return /*#__PURE__*/React.createElement(View, {
    style: [styles.buttonContainer, {
      maxWidth: buttonsMaxSize
    }]
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => goToNewSlide(active + 1, slide, setSlide, numberOfSlide, onDone, navContainerMaxSize, dotWidth, deviceMaxWidth)
  }, !isLastSlide ? /*#__PURE__*/React.createElement(Animated.View, {
    style: {
      opacity: _opacityOfNextButton
    }
  }, renderNextButton()) : /*#__PURE__*/React.createElement(Animated.View, {
    style: {
      opacity: _opacityOfDoneButton
    }
  }, renderDoneButton())));
};

const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
export default NextContainer;
