// react-native-slider-intro/lib/module/components/PrevContainer.js
import React from 'react';
import { View, TouchableOpacity, Animated } from 'react-native';
import { StyleSheet } from 'react-native';

const handlePress = (onSkip, setDefaultState, type, goToNewSlide, slide, setSlide, numberOfSlide, onDone, navContainerMaxSize, dotWidth, deviceMaxWidth) => {
  const {
    active
  } = slide;

  if (type === 'skip') {
    setDefaultState();
    onSkip();
    return;
  }

  goToNewSlide(active - 1, slide, setSlide, numberOfSlide, onDone, navContainerMaxSize, dotWidth, deviceMaxWidth);
};

const PrevContainer = ({
  setDefaultState,
  onSkip,
  buttonsMaxSize,
  _opacityOfSkipButton,
  renderSkipButton,
  type,
  goToNewSlide,
  slide,
  setSlide,
  numberOfSlide,
  onDone,
  navContainerMaxSize,
  dotWidth,
  deviceMaxWidth,
  showLeftButton
}) => {
  return /*#__PURE__*/React.createElement(View, {
    style: [styles.buttonContainer, {
      maxWidth: buttonsMaxSize
    }]
  }, showLeftButton && /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => handlePress(onSkip, setDefaultState, type, goToNewSlide, slide, setSlide, numberOfSlide, onDone, navContainerMaxSize, dotWidth, deviceMaxWidth)
  }, /*#__PURE__*/React.createElement(Animated.View, {
    style: {
      maxWidth: buttonsMaxSize,
      opacity: _opacityOfSkipButton
    }
  }, renderSkipButton())));
};

const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
export default PrevContainer;
